src/Utilities/Compiler/Extensions/OperationBlockAnalysisContextExtension.cs

```diff
@@ -53,10 +53,16 @@ static bool IsSingleStatementBody(IBlockOperation body)
                     descendants[0] is IThrowOperation throwOperation &&
                     throwOperation.GetThrownExceptionType() is ITypeSymbol createdExceptionType)
                 {
-                    if (Equals(context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemNotImplementedException), createdExceptionType.OriginalDefinition) ||
-                        Equals(context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemNotSupportedException), createdExceptionType.OriginalDefinition) ||
+                    if (SymbolEqualityComparer.Default.Equals(
+                            context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemNotImplementedException),
```

Comment:
Out of scope for this PR, but I'm worried about the performance when `GetOrCreateTypeByMetadataName` is called multiple times in this extension.

I think it may be better to get the symbols once at compilation start for the analyzers using this extension, and pass them to the method.

Reply:
I think that perf should be ok because we are doing a dictionary call (if it exists we return the value otherwise we create, store and return). We implemented this helper especially to help with performance issue.

Reply:
Both of you are right to an extent - perf is much better even inside operation callbacks because it is a dictionary, however we still prefer to collect all well known symbols once at compilation start and pass them around, primary for consistency in our code base and also avoid large number of operation callbacks trying to access the same dictionary when multiple analyzers are executing in parallel.

Reply:
Agreed this is our of scope for this PR though.